atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AdvisoryLockConditionSuppliers.java

```diff
@@ -76,7 +76,16 @@ private static HeldLocksToken acquireLock(LockService lockService, LockRequest l
             if (response == null) {
                 RuntimeException ex = new LockAcquisitionException(
                         "Failed to lock using the provided lock request: " + lockRequest);
-                log.warn("Could not lock successfully", ex);
+                switch (lockRequest.getBlockingMode()) {
```

Comment:
nit: Put in a separate method called "handleCouldNotLock"

Reply:
Now that I also handle failureCount (and logging when we surpass the limit), it's harder to extract. Thoughts?

Reply:
Well now it looks like you changed semantics? Previously you'd always retry, now you won't retry on DO_NOT_BLOCK (it feels correct, but I'm surprised there isn't a test that's failing now?)

Reply:
Agreed that this changes semantics.. do you agree that these semantics are indeed what we want or were the old semantics somehow preferable?

Reply:
The new semantics are what I would expect and prefer overall. The concern is that a user using `DO_NOT_BLOCK` might end up failing more often, but I don't think that is actively a problem as previously these attempts would be clustered, and in the worst case the transaction will just need to try again.